misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -258,6 +254,23 @@ internal class RealTransacter private constructor(
     }
   }
 
+  private fun mapTidbException(e: Throwable): Throwable {
```

Comment:
Superficial feedback: you could make this an extension: 

```
private fun Throwable.mapTidbException(): Throwable = when {
 ...
}
```

Then line 239 becomes
```
if (config.type == DataSourceType.TIDB) {
  rethrow = e.mapTidbException()
}
```

Reply:
Making it an extension doesn't seem to read any better?